File: frontend/ediary_cli/src/main/kotlin/org/srd/ediary_cli/model/Owner.kt
package org.srd.ediary_cli.model

import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable

import org.srd.ediary_cli.util.LocalDateSerializer

@Serializable
data class OwnerLoginDTO(val login: String,
                         val password: String)

@Serializable
data class OwnerInfoDTO(val id: Long,
                        val name: String,
                        @Serializable(with = LocalDateSerializer::class)
                        val birthDate: LocalDate,
                        val login: String,
                        @Serializable(with = LocalDateSerializer::class)
                        val createdDate: LocalDate)

@Serializable
data class OwnerCreateDTO(val name: String,
                          @Serializable(with = LocalDateSerializer::class)
                          val birthDate: LocalDate,
                          val login: String,
                          val password: String)

File: frontend/ediary_vue/src/types/Owner.ts
export interface Owner {
    id: number;
    name: string;
    birthDate: string;
    login: string;
    password: string;
    createdDate: string;
}

export interface OwnerLoginDTO {
    login: string;
    password: string;
}

export interface OwnerInfoDTO {
    id: number;
    name: string;
    birthDate: string;
    login: string;
    createdDate: string;
}

export interface OwnerCreateDTO {
    name: string;
    birthDate: string;
    login: string;
    password: string;
}

export interface TokenResponse {
    token: string;
}

export interface TokenRequest {
    login: string;
    password: string;
}
